### ISSUE TITLE
save the aggregated command of the run

### ISSUE THREAD

**Issue by lmoffatt** [lmoffatt/macro_dr#51]
The problem is that this is a self referential command. We have to save the aggregate the commands, extracts from them the Run Id and then save them. The best place to do this is in the command run.

**Comment by lmoffatt** [comment]
So, the program has to have a representation of itself, that it.

**Comment by lmoffatt** [comment]
I tried this 
```++c
s="write_text(filename=runIdName, text=""" +s+ """\n"+s;
```

It fails miserably. 
I will try now another hack: saving the string program in a temporary and making the program to rename the file.

**Comment by lmoffatt** [comment]
Ok, a second try. 
I implemented the command write_script(script_name=runIdName) that saves current script. 
For that purpose it uses a temporary file :
```c++
write_text(temp_script_file,s);
 ```
where ```s``` is the string with the whole script. 
So it is a "dirty" hack, a proper solution demands a reimplementation of dcli with a proper reflection grammar. 
implemented on commit 62722e7
